### ISSUE TITLE
Question about coordinate system 

### ISSUE THREAD

**Issue by jh27kim** [ashawkey/stable-dreamfusion#196]
Hi, I'm currently working on a project using SDS. 
Seems like you are using left-handed camera system where Rotation matrix is calculated as below.

# provider.py 
right_vector = safe_normalize(torch.cross(forward_vector, up_vector, dim=-1)) 

and in ray sampling, z is applied when converting to homogeneous coordinates. 
# utils.py
zs = - torch.ones_like(i)
xs = - (i - cx) / fx * zs
ys = (j - cy) / fy * zs

So my question is, 
i) Are you using left-handed system? If I were to fix it to right-handed system, is this the only part that needs modification? (Maybe it also affects shading, etc?)

ii) Why zs is applied to xs, ys? and -1 to xs? 

Thanks,

**Comment by jh27kim** [comment]
Yes, thank you for pointing that out. 
I have fixed left handed to right-handed system by converting the order of cross product. 

But it seems to me that xs and ys should also be changed as:
BEFORE
zs = - torch.ones_like(i)
xs = - (i - cx) / fx * zs
ys = (j - cy) / fy * zs

AFTER
zs = - torch.ones_like(i)
xs = (i - cx) / fx
ys =  (H-1 - j - cy) / fy 

to match right-handed coordinate system 

up vector (+y)
   |
   |
   |
   |__________> right vector (+x)
  /
 / 
/
forward vector(+z)

**Comment by lingtengqiu** [comment]
> 

Note zs is -1, hence 
```xs = - (i - cx) / fx * zs
 ys = (j - cy) / fy * zs
```
is equal to 
```xs = (i - cx) / fx
ys = -(j - cy) / fy
```

Note that the author use left-hand-side opengl, using the above equations results in  left-hand-side opencv coordinate. Note that in the left-hand-side opencv coordinate, NeRF renders image aligned with meshlab mesh.

**Comment by lingtengqiu** [comment]
I conduct right-hand side experiments, and the right-hand side's results also work well as left-hand side.

### left-hand-side
![Screenshot from 2023-04-13 11-23-52](https://user-images.githubusercontent.com/37240099/231640966-bde1c4f8-b4d1-4cf9-9b28-1d13fca0802e.png)

### right-hand-side

![Screenshot from 2023-04-13 11-23-59](https://user-images.githubusercontent.com/37240099/231641152-95fa5969-2f23-4371-9c64-bbfeeb8d9ef7.png)

**Comment by jh27kim** [comment]
I see thank you for clarifying ! 

One last thing is wouldn't it make more sense to modify 
ys = -(j - cy) / fy     ->.    ys = (H-1 - j - cy) / fy ?

**Comment by lingtengqiu** [comment]
> I see thank you for clarifying !
> 
> One last thing is wouldn't it make more sense to modify ys = -(j - cy) / fy ->. ys = (H-1 - j - cy) / fy ?

broadly speaking, cy = H/2.
